### PR TITLE
feat(autogpt_builder): Enable loading an existing flow

### PR DESCRIPTION
**Pull request by Pwuts**
### Background

* Needed for the "Edit flow" button of #7335

### Changes 🏗️

- Add `flowID` query parameter
- Add logic to load a flow from the server into the builder

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [x] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [x] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [x] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

File: rnd/autogpt_builder/src/components/Flow.tsx [hunk 1]
```diff
@@ -153,60 +163,98 @@ const Flow: React.FC = () => {
...
+        delete inputData[key];
+      }
+    });
 
-  console.log(`Final prepared input for ${node.data.blockType} (${node.id}):`, inputData);
-  return inputData;
-};
+    console.log(`Final prepared input for ${node.data.blockType} (${node.id}):`, inputData);
+    return inputData;
+  };
```
Comment (Pwuts) [inline rnd/autogpt_builder/src/components/Flow.tsx:255]: this is a very strange diff render of a simple indentation fix

Comment (Pwuts) [reply rnd/autogpt_builder/src/components/Flow.tsx:255]: I can revert these changes if it helps to get this merged

File: rnd/autogpt_builder/src/app/page.tsx [hunk 2]
```diff
@@ -32,7 +34,7 @@ export default function Home() {
 
           <div className="w-full flex justify-center mt-10">
               <div className="flow-container w-full h-full">
-                  <Flow/>
+                  <FlowEditor flowID={useSearchParams().get("flowID") ?? undefined} />
```
Comment (aarushik93) [inline rnd/autogpt_builder/src/app/page.tsx:37]: renaming will cause conflicts in the current WIPs - do you think this is 100% necessary for the functionality in this PR or can the rename be introduced in a different PR? - just wanna be more mindful rn and minimize the conflicts each of us are gonna have early on

File: rnd/autogpt_builder/src/components/Flow.tsx (hunk 1 above)
Comment (aarushik93) [reply rnd/autogpt_builder/src/components/Flow.tsx:255]: nah its okay, we should get the indentation fixes in, itll become more frustrating down the line

File: rnd/autogpt_builder/src/app/page.tsx (hunk 2 above)
Comment (Pwuts) [reply rnd/autogpt_builder/src/app/page.tsx:37]: I have to rename *something*. Otherwise the `Flow` type and `Flow` component clash. I thought it made more sense to rename `<Flow>` to better represent its functionality.

Comment (aarushik93) [reply rnd/autogpt_builder/src/app/page.tsx:37]: I was thinking the type should be called graph since it's an API representation. But i don't think it's a blocker, just an idea

**Review by aarushik93 — APPROVED** [review]
Linting is failing, but after that's fixed LGTM

File: rnd/autogpt_builder/src/app/page.tsx (hunk 2 above)
Comment (Pwuts) [reply rnd/autogpt_builder/src/app/page.tsx:37]: I guess we could do that. But that would also require a refactor of the API client for consistency. Could do later in a PR if that's something we want.

**Comment by Pwuts** [comment]
> Linting is failing, but after that's fixed LGTM

I can't fix all of the errors:
```
79:6  Warning: React Hook useEffect has a missing dependency: 'api'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
```
This isn't a change I made; the dependency list is empty in `master`. Fixing this makes it loop endlessly.

```
88:8  Warning: React Hook useEffect has missing dependencies: 'api' and 'loadFlow'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps
```
Same as above, adding these to the dependency list makes the effect run repeatedly, endlessly.

Fixing this would require not initializing/defining `api` and `loadFlow` in the component render function.
